### PR TITLE
fix(gorgone): Add missing centengine report part to centreon-storage database

### PR DESCRIPTION
**Pull request by kduret**
## Description

Migrated from https://github.com/centreon/centreon/pull/4465

some of the poller performance report is missing since the gorgone migration.
This pr aim to add all data that was previously added in centreon-storage.

Fixes # MON-37990

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>

on central, launch gorgone with statistic module enabled (default configuration) 
wait 5 to 10 minutes for statistics job to be run (see gorgone log
generate a poller performance report and check all data are present, or check in db  : 
here an exemple with one central and one poller : 
```
MariaDB [centreon-storage]> select * from nagios_stats;
+-------------+----------+------------+------------------------------+
| instance_id | stat_key | stat_value | stat_label                   |
+-------------+----------+------------+------------------------------+
|           1 | Min      | 0.102      | Service Check Latency        |
|           1 | Max      | 0.955      | Service Check Latency        |
|           1 | Average  | 0.550      | Service Check Latency        |
|           1 | Min      | 0.001      | Service Check Execution Time |
|           1 | Max      | 0.332      | Service Check Execution Time |
|           1 | Average  | 0.132      | Service Check Execution Time |
|           1 | Min      | 0.020      | Host Check Latency           |
|           1 | Max      | 0.868      | Host Check Latency           |
|           1 | Average  | 0.475      | Host Check Latency           |
|           1 | Min      | 0.030      | Host Check Execution Time    |
|           1 | Max      | 0.152      | Host Check Execution Time    |
|           1 | Average  | 0.083      | Host Check Execution Time    |
|           2 | Min      | 0.102      | Service Check Latency        |
|           2 | Max      | 0.955      | Service Check Latency        |
|           2 | Average  | 0.550      | Service Check Latency        |
|           2 | Min      | 0.001      | Service Check Execution Time |
|           2 | Max      | 0.332      | Service Check Execution Time |
|           2 | Average  | 0.132      | Service Check Execution Time |
|           2 | Min      | 0.020      | Host Check Latency           |
|           2 | Max      | 0.868      | Host Check Latency           |
|           2 | Average  | 0.475      | Host Check Latency           |
|           2 | Min      | 0.030      | Host Check Execution Time    |
|           2 | Max      | 0.152      | Host Check Execution Time    |
|           2 | Average  | 0.083      | Host Check Execution Time    |
+-------------+----------+------------+------------------------------+

```
#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by mushroomempires — APPROVED** [review]

**Review by omercier — APPROVED** [review]

**Review by sdepassio — APPROVED** [review]

**Comment by tuntoja** [comment]
shouldn t tests be green ?

**Review by tuntoja — APPROVED** [review]

**Review by omercier — APPROVED** [review]

File: .github/docker/Dockerfile.gorgone-testing-bookworm [hunk 1]
```diff
@@ -2,27 +2,30 @@ FROM debian:bookworm
...
-    echo "deb https://packages.centreon.com/apt-standard-24.05-testing/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/centreon.list && \
-    echo "deb https://packages.centreon.com/apt-plugins-testing/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/centreon-plugins.list && \
-    wget -O- https://apt-key.centreon.com | gpg --dearmor | tee /etc/apt/trusted.gpg.d/centreon.gpg > /dev/null 2>&1 && \
-    apt-get clean && rm -rf /var/lib/apt/lists/*
\ No newline at end of file
+RUN bash -e <<EOF
+
+apt-get update
+
+# Install required lib for language and apt installation. zstd is required for cache management in GHA
```
Comment (smau-centreon) [inline .github/docker/Dockerfile.gorgone-testing-bookworm:9]:
```suggestion
# Install the required lib for language and apt installation. zstd is required for cache management in GHA.
```

File: .github/docker/Dockerfile.gorgone-testing-bookworm [hunk 2]
```diff
@@ -2,27 +2,30 @@ FROM debian:bookworm
...
+# Install Robotframework and tests dependency
```
Comment (smau-centreon) [inline .github/docker/Dockerfile.gorgone-testing-bookworm:10]:
```suggestion
# Install Robotframework and test dependency
```

File: .github/docker/Dockerfile.gorgone-testing-bullseye [hunk 3]
```diff
@@ -1,30 +1,28 @@
...
 
-apt-get update
-apt-get install -y locales libcurl4-openssl-dev curl wget zstd jq lsb-release mariadb-client iproute2
-rm -rf /var/lib/apt/lists/*
-localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
-apt-get clean
+apt update
 
-EOF
+# Install required lib for language and apt installation. zstd is required for cache management in GHA
```
Comment (smau-centreon) [inline .github/docker/Dockerfile.gorgone-testing-bullseye:11]:
```suggestion
# Install the required lib for language and apt installation. zstd is required for cache management in GHA.
```

File: .github/docker/Dockerfile.gorgone-testing-bullseye [hunk 4]
```diff
@@ -1,30 +1,28 @@
...
+# Install Robotframework and tests dependency
```
Comment (smau-centreon) [inline .github/docker/Dockerfile.gorgone-testing-bullseye:12]:
```suggestion
# Install Robotframework and test dependency
```

File: gorgone/docs/modules/centreon/statistics.md [hunk 5]
```diff
@@ -2,15 +2,15 @@
 
 ## Description
 
-This module aims to deal with statistics collection of Centreon Engine and Broker.
+This module aims to deal with statistics collection of Centreon Engine and Broker. It require action module to be configured on each poller and on the central.
```
Comment (smau-centreon) [inline gorgone/docs/modules/centreon/statistics.md:5]:
```suggestion
This module aims to manage the collection of Centreon Engine and Broker statistics. It requires the configuration of an action module on each poller and on the central.

```

File: gorgone/docs/modules/centreon/statistics.md [hunk 6]
```diff
@@ -2,15 +2,15 @@
...
 
 ## Configuration
 
 | Directive        | Description                                                                                    | Default value                     |
 | :--------------- | :--------------------------------------------------------------------------------------------- | :-------------------------------- |
 | broker_cache_dir | Path to the Centreon Broker statistics directory (local) use to store node's broker statistics | `/var/lib/centreon/broker-stats/` |
 
-The configuration needs a cron definition to unsure that statistics collection will be done cyclically.
+The configuration needs a cron definition to unsure that statistics collection will be done regulary.
```
Comment (smau-centreon) [inline gorgone/docs/modules/centreon/statistics.md:13]:
```suggestion
The configuration needs a cron definition to ensure that statistics are collected regularly.

```

File: gorgone/docs/modules/centreon/statistics.md [hunk 7]
```diff
@@ -30,13 +30,32 @@ cron:
...
+| STATISTICSREADY    | Internal event to notify the core                 | 
+| STATISTICSLISTENER | Internal Event to receive data from action module |
+| BROKERSTATS        | Collect Centreon Broker statistics files on node  |
+| ENGINESTATS        | Collect Centreon engine statistics on node        |
 
 ## API
 
+### Collect Centreon engine statistics on every nodes configured
+
+The api send back a token to monitor advancement. Please note this token don't allow to monitor the whole process, only the first part until action command are sent.
```
Comment (smau-centreon) [inline gorgone/docs/modules/centreon/statistics.md:44]:
```suggestion
The API returns a token to monitor the advancement. Please note this token does not allow to monitor the whole process but only the first part until an action command is sent.

```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

File: gorgone/docs/modules/centreon/statistics.md [hunk 8]
```diff
@@ -2,15 +2,15 @@
 
 ## Description
 
-This module aims to deal with statistics collection of Centreon Engine and Broker.
+This module aims to manage the collection of Centreon Engine and Broker statistics. It require requires the configuration of an action module on each poller and on the central.
```
Comment (smau-centreon) [inline gorgone/docs/modules/centreon/statistics.md:5]:
```suggestion
This module aims to manage the collection of Centreon Engine and Broker statistics. It requires the configuration of an action module on each poller and on the central.

```

File: gorgone/docs/modules/centreon/statistics.md [hunk 9]
```diff
@@ -2,15 +2,15 @@
...
 
 ## Configuration
 
 | Directive        | Description                                                                                    | Default value                     |
 | :--------------- | :--------------------------------------------------------------------------------------------- | :-------------------------------- |
 | broker_cache_dir | Path to the Centreon Broker statistics directory (local) use to store node's broker statistics | `/var/lib/centreon/broker-stats/` |
 
-The configuration needs a cron definition to unsure that statistics collection will be done cyclically.
+The configuration needs a cron definition to ensure that statistics collection are collected regularly.
```
Comment (smau-centreon) [inline gorgone/docs/modules/centreon/statistics.md:13]:
```suggestion
The configuration needs a cron definition to ensure that statistics are collected regularly.

```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**